SlowCheetah.VisualStudio/NugetHandler/SlowCheetahNuGetManager.cs

```diff
@@ -0,0 +1,191 @@
+            if (!string.IsNullOrEmpty(propertyValue))
+            {
+                return true;
+            }
+
+            buildPropertyStorage.GetPropertyValue("SlowCheetahTargets", null, (uint)_PersistStorageType.PST_PROJECT_FILE, out propertyValue);
+            if (!string.IsNullOrEmpty(propertyValue))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool IsSlowCheetahInstalled(Project project)
+        {
+            var componentModel = (IComponentModel)this.package.GetService(typeof(SComponentModel));
+            IVsPackageInstallerServices installerServices = componentModel.GetService<IVsPackageInstallerServices>();
+            return installerServices.IsPackageInstalled(project, PackageName);
+        }
+
+        private bool IsSlowCheetahUpdated(Project project)
+        {
+            // Checks for older SC versions that require more complex update procedure
+            var componentModel = (IComponentModel)this.package.GetService(typeof(SComponentModel));
+            IVsPackageInstallerServices installerServices = componentModel.GetService<IVsPackageInstallerServices>();
+            IVsPackageMetadata scPackage =
+                    installerServices.GetInstalledPackages().First(pkg => string.Equals(pkg.Id, PackageName, StringComparison.OrdinalIgnoreCase));
+            if (scPackage != null)
+            {
+                Version ver;
+                if (Version.TryParse(scPackage.VersionString, out ver))
+                {
+                    return ver > new Version(2, 5, 15);
+                }
+            }
+
+            return false;
+        }
+
+        private void BackgroundInstallSlowCheetah(Project project)
+        {
+            string projName = project.UniqueName;
+            Task installTask;
+            if (!this.installTasks.TryGetValue(projName, out installTask))
+            {
+                if (this.HasUserAcceptedWarningMessage())
+                {
+                    // Gets the general output pane to inform user of installation
+                    IVsOutputWindowPane outputWindow = (IVsOutputWindowPane)this.package.GetService(typeof(SVsGeneralOutputWindowPane));
+                    if (outputWindow != null)
+                    {
+                        outputWindow.OutputString(string.Format("Installing SlowCheetah NuGet package to {0} ...\n", project.Name));
+                    }
+
+                    // Installs the latest version of the SlowCheetah NuGet package
+                    var componentModel = (IComponentModel)this.package.GetService(typeof(SComponentModel));
+                    IVsPackageInstaller2 packageInstaller = componentModel.GetService<IVsPackageInstaller2>();
+                    installTask = Task.Run(() =>
+                    {
+                        InstallSlowCheetahPackage(packageInstaller, project);
+                    }).ContinueWith(t =>
```

Comment:
Don't need a `ContinueWith` here. Having this logic inside the `Task.Run` above is fine.